### PR TITLE
Require Approver Comment and Allow Updating Approver Comments

### PR DESCRIPTION
**Pull request by Colin-Schultz**

### CODE REVIEW

File: services/rds/src/main/java/org/finra/gatekeeper/services/accessrequest/AccessRequestService.java [hunk 1]
```diff
@@ -509,6 +525,19 @@ public List<ActiveAccessRequestWrapper> cancelRequest(String taskId, Long reques
...
 
+    /**
+     * Updates the approver comment of the Request
+     * @param taskId - the activiti task id
+     * @param requestId - the AccessRequest object id
+     * @param approverComments - The comments from the approver
+     * @return - The updated list of Active Access Requests
+     */
+    @PreAuthorize("@gatekeeperRoleService.isApprover()")
+    public List<ActiveAccessRequestWrapper> updateRequestApproverComment(String taskId, Long requestId, String approverComments ) {
```
Comment (r331) [inline services/rds/src/main/java/org/finra/gatekeeper/services/accessrequest/AccessRequestService.java:536]: unused variable - taskId

File: services/rds/src/main/java/org/finra/gatekeeper/controllers/AccessRequestController.java [hunk 2]
```diff
@@ -109,6 +109,11 @@ public Object cancelRequest(@RequestBody ActiveAccessRequestWrapper request) {
         return accessRequestService.cancelRequest(request.getTaskId(), request.getId());
     }
 
+    @RequestMapping(value = "/updateRequestApproverComment", method = RequestMethod.PUT, produces = MediaType.APPLICATION_JSON_VALUE)
```
Comment (SMxJrz) [inline services/rds/src/main/java/org/finra/gatekeeper/controllers/AccessRequestController.java:112]: I Know we're only going to be allowing the request approver comment at this point Can we keep the name of this more general? like "updateRequest"

File: services/rds/src/main/java/org/finra/gatekeeper/services/accessrequest/AccessRequestService.java [hunk 3]
```diff
@@ -509,6 +509,23 @@ public List<ActiveAccessRequestWrapper> cancelRequest(String taskId, Long reques
         return getActiveRequests();
     }
 
+    /**
+     * Updates the approver comment of the Request
+     * @param requestId - the AccessRequest object id
+     * @param approverComments - The comments from the approver
+     * @return - The updated list of Active Access Requests
+     */
+    public List<ActiveAccessRequestWrapper> updateRequestApproverComment(Long requestId, String approverComments ) {
```
Comment (SMxJrz) [inline services/rds/src/main/java/org/finra/gatekeeper/services/accessrequest/AccessRequestService.java:518]: comment in Controller class applies here too

File: ui/app/component/shared/RequestDataService.js [hunk 4]
```diff
@@ -31,6 +31,7 @@ class RequestDataService extends DataService{
         this.rejectRequest = 'rejectRequest';
         this.cancelRequest = 'cancelRequest';
         this.getRequests = 'requests';
+        this.updateRequest = 'updateRequestApproverComment';
```
Comment (SMxJrz) [inline ui/app/component/shared/RequestDataService.js:34]: same as other comment

File: ui/app/component/shared/RequestDataService.js [hunk 5]
```diff
@@ -57,6 +58,9 @@ class RequestDataService extends DataService{
         return this.http.put(this.getApi()+"/"+this.cancelRequest, request);
     }
 
+    updateRequestApproverComment(request){
```
Comment (SMxJrz) [inline ui/app/component/shared/RequestDataService.js:61]: same as other comment

**Review by SMxJrz — CHANGES_REQUESTED** [review]

**Review by SMxJrz — APPROVED** [review]

**Pull request merged**